tools/c7n_gcp/c7n_gcp/query.py

```diff
@@ -211,21 +211,19 @@ def _fetch_resources(self, query):
         try:
             return self.augment(self.source.get_resources(query)) or []
         except HttpError as e:
-            error_reason, error_code, error_message = extract_errors(e)
+            error_code, error_message = extract_errors(e)
 
-            if error_reason is None and error_code is None:
+            if error_code is None and error_message is None:
                 raise
-            if error_code == 403 and 'disabled' in error_message:
-                log.warning(error_message)
-                return []
-            elif error_reason == 'accessNotConfigured':
+            else:
                 log.warning(
-                    "Resource:%s not available -> Service:%s not enabled on %s",
+                    "%s: %s -> Resource:%s Service:%s Project:%s",
```

Comment:
I dig the approach but still noodling on the content here, there's possibility of getting a null err code or message here, and clarifying perhaps that resources are not available (ie this occurred when fetching resources), and possibly also the error itself.

Reply:
Thanks, I am banking on the fact that the GCP resource APIs will return at least an error and message. From what I saw you always get both of these back but if it does not then I do raise the exception since something funky has happened. There is an argument for adding more information to the message when there is only an error code but then I am not sure where I would get the extra information from.

Another option could be to add something along the lines of "Error occurred while querying resources" to the `log.warning` for clarification that an error did occur. 

It could also be an option to switch to `log.error` for anything that isn't a 400 response, since at that point something probably needs to be fixed, and keep it the same for 400s?

Reply:
i'm going to push some minor formatting changes, basically shifting the error code / message to the end since the message itself is potentially unbounded, to make scanning the logs a bit easier. For 500 errors stopping execution of the policy seems reasonable, each policy execution to project should be fully independent, ie. an error on one shouldn't affect others.

with the change the formatting looks like
```
2020-12-01 12:07:24,542: googleapiclient.http:WARNING Encountered 403 Forbidden with reason "PERMISSION_DENIED"
2020-12-01 12:07:24,543: c7n_gcp.query:WARNING Project:rack-sam Resource:function unavailable Service:cloudfunctions Error:403 
 Message:Cloud Functions API has not been used in project redacted before or it is disabled. Enable it by visiting https://console.developers.google.com/apis/api/cloudfunctions.googleapis.com/overview?project=redacted then retry. If you enabled this API recently, wait a few minutes for the action to propagate to our systems and retry.
```